### PR TITLE
1 Extract writing to svg file to separate entities

### LINKED ISSUE #1: Extract writing to SVG file to separate entities
- Extract lines of code responsible to generate SVG/XML file to separate entities.
- Introduce first version of storing properties

### PR DESCRIPTION
**Pull request by Nutek**
- Collection of helper object for Tag-Attribute-Tree handling
- Replacement of inline code formatting.

### CODE REVIEW

**Comment by kris039** [comment]
Add test files to subfolder to increase clarity

File: test_helpers.py
Comment (kris039) [inline test_helpers.py:1]: Misleading file name

Comment (kris039) [reply test_helpers.py:1]: "testing_helpers" or something

File: test_tags_attributes_tree_format.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        [
+            ({"attr1": "val1"}, ['<element attr1="val1" />']),
+            ({"attr2": "val2"}, ['<element attr2="val2" />']),
+            (
+                {"attr1": "val1", "attr2": "val2"},
+                ['<element attr1="val1" attr2="val2" />'],
+            ),
+        ],
+    )
+    def test_empty_element_with_attributes(self, attributes, expected):
```
Comment (kris039) [inline test_tags_attributes_tree_format.py]: Add test cases for:
- empty strings
- `None` values
- numbers
- special characters

File: test_tags_attributes_tree_format.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        assert get_trimmed_lines(node) == expected
+
+    @pytest.mark.parametrize(
+        "label,expected",
+        [
+            ("comment", ["<!-- comment -->"]),
+            ("other text", ["<!-- other text -->"]),
+        ],
+    )
+    def test_some_comment(self, label, expected):
```
Comment (kris039) [inline test_tags_attributes_tree_format.py]: Add cases for:
- `None`
- empty strings
- comment ending sequence

File: test_tags_attributes_tree_format.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+        node = TAT_Node("root")
+        node.add_nodes(TAT_Node("inner1"), TAT_Node("inner2"))
+        assert get_trimmed_lines(node) == [
+            "<root>",
+            "<inner1 />",
+            "<inner2 />",
+            "</root>",
+        ]
+
+    def test_adding_multiple_attributes(self):
```
Comment (kris039) [inline test_tags_attributes_tree_format.py]: Add cases for:
- empty string
- `None`

File: test_tags_attributes_tree_format.py [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+        nested.add_node(TAT_Node("inner"))
+        nested.add_node(TAT_Comment("text"))
+        node.add_node(nested)
+        assert (
+            str(node)
+            == "<root>\n <inner />\n <nested>\n  <inner />\n  <!-- text -->\n </nested>\n</root>"
+        )
+
+
+class TestTagsAttributeFormatMultipleAddition:
```
Comment (kris039) [inline test_tags_attributes_tree_format.py]: Add case empty parameter

File: test_tags_attributes_tree_format.py [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+            '<root name:attr1="val1" name:attr2="val2" />',
+        ]
+
+
+class TestTagsAttributeFormatChaining:
+    def test_adding_nodes_by_chaining(self):
+        node = (
+            TAT_Node("root")
+            .add_node(TAT_Node("inner1"))
+            .add_nodes(TAT_Node("inner2"))
```
Comment (kris039) [inline test_tags_attributes_tree_format.py]: Add separate cases for add_node and add_nodes methods

File: test_tags_attributes_tree_format.py [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+            ("other text", ["<!-- other text -->"]),
+        ],
+    )
+    def test_some_comment(self, label, expected):
+        node = TAT_Comment(label)
+        assert get_trimmed_lines(node) == expected
+
+
+class TestTagsAttributeFormatNesting:
+    def test_contains_empty_element(self):
```
Comment (kris039) [inline test_tags_attributes_tree_format.py]: Add cases for:
- invalid object (ex. string)
- `None`

File: test_tags_attributes_tree_format.py [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+        )
+        assert get_trimmed_lines(node) == [
+            "<root>",
+            "<inner1 />",
+            "<inner2 />",
+            "<inner3 />",
+            "</root>",
+        ]
+
+    def test_adding_attributes_by_chaining(self):
```
Comment (kris039) [inline test_tags_attributes_tree_format.py]: Add separate cases for add_attr and add_attrs methods

File: export_svg_280.py [hunk 8]
```diff
@@ -1,1141 +1,2121 @@
...
+        formated_children = [c.format_string(spacer + 1) for c in self._children]
+        return "\n".join(
+            [
+                f"{spacer}<{self._name}{formated_attributes}>",
+                *formated_children,
+                f"{spacer}</{self._name}>",
+            ]
+        )
+
+    def add_node(self, tag):
```
Comment (kris039) [inline export_svg_280.py]: Add name validation

File: export_svg_280.py [hunk 9]
```diff
@@ -1,1141 +1,2121 @@
...
+        self._children.append(tag)
+        return self
+
+    def add_nodes(self, *tags):
+        for tag in tags:
+            self.add_node(tag)
+        return self
+
+    def add_attr(self, name: str, value):
```
Comment (kris039) [inline export_svg_280.py]: Add name validation

**Review by kris039 — CHANGES_REQUESTED** [review]

File: test_helpers.py
Comment (Nutek) [reply test_helpers.py:1]: Done

File: test_tags_attributes_tree_format.py (hunk 7 above)
Comment (Nutek) [reply test_tags_attributes_tree_format.py]: Method `add_attr` removed

File: test_tags_attributes_tree_format.py (hunk 5 above)
Comment (Nutek) [reply test_tags_attributes_tree_format.py]: Method `add_node` removed

File: test_tags_attributes_tree_format.py (hunk 1 above)
Comment (Nutek) [reply test_tags_attributes_tree_format.py]: Done

File: test_tags_attributes_tree_format.py (hunk 3 above)
Comment (Nutek) [reply test_tags_attributes_tree_format.py]: Due to removed method `add_attr` those cases are covered by different tests. Please refer to `test_element_raises_exception_for_invalid_type_of_attribute_name` or `test_element_raises_exception_for_invalid_type_of_attribute_value`

File: test_tags_attributes_tree_format.py (hunk 4 above)
Comment (Nutek) [reply test_tags_attributes_tree_format.py]: Done.

File: test_tags_attributes_tree_format.py (hunk 6 above)
Comment (Nutek) [reply test_tags_attributes_tree_format.py]: Done

File: export_svg_280.py (hunk 8 above)
Comment (Nutek) [reply export_svg_280.py]: Done

File: export_svg_280.py (hunk 9 above)
Comment (Nutek) [reply export_svg_280.py]: Done

File: test_tags_attributes_tree_format.py (hunk 2 above)
Comment (Nutek) [reply test_tags_attributes_tree_format.py]: Done

**Review by kris039 — APPROVED** [review]

**Pull request merged**